Sources/UI/SwiftUI/Views/RemoteImage/RemoteImage.swift

```diff
@@ -32,38 +30,45 @@ public struct RemoteImage<Placeholder: View>: View {
   private let url: URL?
   private let animated: Bool
   private var placeholder: Placeholder?
+  private var onSuccess: ((RetrieveImageResult) -> Void)?
+  private var onError: ((KingfisherError) -> Void)?
   
-  public init(url: URL?, animated: Bool = false) where Placeholder == EmptyView {
+  public init(
+    url: URL?,
+    animated: Bool = false,
+    onSuccess: ((RetrieveImageResult) -> Void)? = nil,
+    onError: ((KingfisherError) -> Void)? = nil
```

Comment:
Maybe we can apply the same API design as `KFImage`?
So that the usage would be:

```swift
RemoteImage()
  .onSuccess {}
  .onFailure {}
```

What do you think?

Reply:
[Agree](https://github.com/povio/PovioKit/pull/356/commits/74b6cfe17abdd2628924c91181edfd8130b6072e)